gammapy/datasets/flux_points.py

```diff
@@ -310,13 +368,82 @@ def flux_pred(self):
 
     def stat_array(self):
         """Fit statistic array."""
+        return self.available_stat_type[self.stat_type]()
+
+    def _stat_array_chi2(self):
+        """Chi2 statistics."""
         model = self.flux_pred()
         data = self.data.dnde.quantity
         try:
-            sigma = self.data.dnde_err
+            sigma = self.data.dnde_err.quantity
         except AttributeError:
-            sigma = (self.data.dnde_errn + self.data.dnde_errp) / 2
-        return ((data - model) / sigma.quantity).to_value("") ** 2
+            sigma = (self.data.dnde_errn + self.data.dnde_errp).quantity / 2
+        return ((data - model) / sigma).to_value("") ** 2
+
+    def _stat_array_profile(self):
+        """Estimate statitistic from interpolation of the likelihood profile."""
+        model = (self.flux_pred() / self.data.dnde_ref).to_value("")
+        stat = np.zeros(model.shape)
+        for idx, interp in enumerate(self._profile_interpolators):
+            stat[idx] = interp(model[idx])
+        return stat
+
+    def _get_valid_profile_interpolators(self):
+        value_scan = self.data.stat_scan.geom.axes["norm"].center
+        shape_axes = self.data.stat_scan.geom._shape[slice(3, None)]
+        interpolators = np.empty(shape_axes, dtype=object)
+        self.mask_valid = np.ones(self.data.dnde.data.shape, dtype=bool)
+        for idx in np.ndindex(shape_axes):
+            stat_scan = np.abs(
+                self.data.stat_scan.data[idx].squeeze()
+                - self.data.stat.data[idx].squeeze()
+            )
+            self.mask_valid[idx] = np.all(np.isfinite(stat_scan))
+            interpolators[idx] = interpolate_profile(
+                value_scan,
+                stat_scan,
+                interp_scale=self.stat_kwargs["interp_scale"],
+                extrapolate=self.stat_kwargs["extrapolate"],
+            )
+        return interpolators
+
+    def _stat_array_distrib(self):
+        """Estimate statistic from probability distributions,
+        assumes that flux points correspond to asymmetric gaussians
+        and upper limits complemantary error functions.
+        """
+        model = self.flux_pred()
+
+        is_ul = self.data.is_ul.data.flatten()
+        n_points = len(is_ul)
+        stat = np.zeros(model.shape)
+        for k in range(n_points):
```

Comment:
Does this work correctly for `FluxPoints` with both `energy` and `time` axes?

Reply:
fixed